firebase_admin/db.py

```diff
@@ -125,40 +125,91 @@ def child(self, path):
         full_path = self._pathurl + '/' + path
         return Reference(client=self._client, path=full_path)
 
-    def get(self):
-        """Returns the value at the current location of the database.
+    def _get_with_etag(self):
+        """Returns the value at the current location of the database, along with its ETag.
+        """
+        data, headers = self._client.request('get', self._add_suffix(),
+                                             headers={'X-Firebase-ETag' : 'true'},
+                                             resp_headers=True)
+        etag = headers.get('ETag')
+        return etag, data
+
+    def get_etag(self):
+        """Returns the ETag at the current location of the database.
 
         Returns:
-          object: Decoded JSON value of the current database Reference.
+            object: ETag of the current database Reference.
+        """
+        headers = self._client.request('get', self._add_suffix(),
+                                       headers={'X-Firebase-ETag' : 'true'},
+                                       resp_headers=True,
+                                       params='print=silent')
+        etag = headers.get('ETag')
+        return etag
+
+    def get(self, etag=False):
+        """Returns the value, and possibly the ETag, at the current location of the database.
+
+        Returns:
+          object: Decoded JSON value of the current database Reference if etag=False, otherwise
+                  the decoded JSON value and the corresponding ETag.
 
         Raises:
           ApiCallError: If an error occurs while communicating with the remote database server.
         """
-        return self._client.request('get', self._add_suffix())
+        if etag:
+            return self._get_with_etag()
+        else:
+            return self._client.request('get', self._add_suffix())
 
-    def _get_with_etag(self):
-        """Returns the value at the current location of the database, along with its ETag."""
-        data, headers = self._client.request(
-            'get', self._add_suffix(), headers={'X-Firebase-ETag' : 'true'}, resp_headers=True)
-        etag = headers.get('ETag')
-        return etag, data
+    def _set_with_etag(self, value, etag):
+        """Sets the data at this location to the specified value, if the etag matches.
+        """
+        if not value or not isinstance(value, dict):
```

Comment:
The value doesn't have to be a `dict` right? I think any JSON-serializable value should work here.